modules/trusted-firmware-m/CMakeLists.txt

```diff
@@ -180,35 +205,34 @@ function(trusted_firmware_build)
     TFM_S_NS_SIGNED_BIN_FILE ${TFM_S_NS_SIGNED_BIN_FILE} # Merged TFM Secure/Nonsecure FW (signed)
     )
 
-  add_library(tfm_api
-    ${ZEPHYR_BASE}/modules/trusted-firmware-m/src/zephyr_tfm_log.c
-  )
+  zephyr_library_named(tfm_api)
```

Comment:
@tejlmand I tried to use zephyr_library_named more correctly here. It works, but I get a warning from Cmake

    CMake Warning at ../../../CMakeLists.txt:710 (message):
      No SOURCES given to Zephyr library: tfm_api

      Excluding target from build.

Reply:
because you should be using `zephyr_library_sources()`, `zephyr_library_sources_ifdef()`, and `zephyr_library_sources_ifndef()` in code below.

Reply:
You mean outside the function?

Reply:
Sorry, I misread. I see what you mean now.

Reply:
Fixed.